### PR TITLE
Add PR Creator for Bitbucket

### PR DESCRIPTION
**Pull request by tooppaaa**
What I Did: 

Use what has been done for Azure by replacing "Azure" with "Bitbucket"

Use API's to 
- Get Branch by name
- Get all PR's
- Create a commit 
- Create PR
- Get all commits

Please note that I'm not a Ruby developer, any feedbacks are welcome !

### CODE REVIEW

**Comment by tooppaaa** [comment]
@jurre I struggle with the formatting due to my ruby knowledge and my setup, any chance we can chat in order to see what's next for this PR ?

**Comment by jarbey** [comment]
+1

**Comment by infin8x** [comment]
Hi @tooppaaa!

Thank you for filing this pull request. Unfortunately, the Dependabot team are currently tied up with some time-sensitive initiatives and we're not able to give this request the attention it needs at the moment. We're hoping to circle back to this and other PRs sometime over the summer.

**Comment by tooppaaa** [comment]
Awesome thanks
We'll keep the fork for now and get our changes back here from time to time !

File: common/lib/dependabot/pull_request_creator/pr_name_prefixer.rb [hunk 1]
```diff
@@ -421,6 +451,14 @@ def codecommit_client_for_source
           )
       end
 
+      def bitbucket_client_for_source
+        @bitbucket_client_for_source ||=
+          Dependabot::Clients::BitbucketTS.for_source(
```
Comment (MMartyn) [inline common/lib/dependabot/pull_request_creator/pr_name_prefixer.rb:456]: This looks like a typo maybe? Not seeing where BitbucketTS is defined.

Comment (tooppaaa) [reply common/lib/dependabot/pull_request_creator/pr_name_prefixer.rb]: Indeed, BitbucketTS is the name I used to fork Bitbucket in the script we're using. Nice catch !

File: common/lib/dependabot/clients/bitbucket.rb [hunk 2]
```diff
@@ -10,12 +10,27 @@ class NotFound < StandardError; end
...
+        new(source, credential)
+      end
+
       ##########
       # Client #
       ##########
 
-      def initialize(credentials:)
-        @credentials = credentials
+      def initialize(source, credential)
```
Comment (renatolond) [inline common/lib/dependabot/clients/bitbucket.rb:31]: This seems to break calls for existing methods (for instance the file fetcher in the generic script).
I think to keep compatibility, it should be something more like
```
def initialize(credentials:, source: nil)
```

**Comment by jurre** [comment]
Hi, apologies on the lack of a response on this. I believe this is a duplicate of https://github.com/dependabot/dependabot-core/pull/1450. Let me paste my reply on that PR, because it applies here as well:

> Hi all, I appreciate the work that went into this PR and this definitely warrants a reply from the maintainers.
> 
> However, at this point I don't think that we can accept this patch into dependabot-core natively, as we don't have the capacity or expertise to provide support for Bitbucket. This is currently already a problem with the existing Azure DevOps and GitLab clients, where we're not positioned well to provide support and rely on other members in the community to do so. However, not having clear maintainers for those parts of the codebase makes this hard.
> 
> Where I would like to see support for (non GitHub) source control providers go in the future, is for them to be stand-alone gems that you can pull in next to dependabot-core. dependabot-core would provide a clear interface for those gems to implement. These gems can then be maintained by people who actually know and use these providers.
> 
> I think this should currently already be possible, but it might rely on monkey-patching some of the existing dependabot-core classes, which is not ideal. I'd be more than happy to accept patches that improve the interface for external providers if you decide to go down that route.

I appreciate the work that went into this, and if anyone wants to tackle releasing this as a standalone gem and there is any support I can provide to make it easier to integrate with dependabot-core I'm happy to help.

**Pull request closed without merging**